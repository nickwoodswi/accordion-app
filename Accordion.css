p.view{
    display: block
}
p.close{
    display: none
}