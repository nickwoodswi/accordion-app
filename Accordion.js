import React, {Component} from 'react'
import './Accordion.css'

const sections = [
    {
      title: 'Section 1',
      content: 'Lorem ipsum dolor sit amet consectetur adipisicing elit.',
    },
    {
      title: 'Section 2',
      content: 'Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!',
    },
    {
      title: 'Section 3',
      content: 'Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?',
    },
]

class Accordion extends Component {
    constructor() {
        super()
        this.state = {
            display: null
        }
    }

    displayContent = (idx) => {
        this.setState({
            display: idx
        })
    }

    render() {
        return(
            sections.map((section, idx) => {
                return (
                    <li key={idx}>
                        <button onClick={() => this.displayContent(idx)}>{section.title}</button>
                        <p className={(idx === this.state.display) ? `view` : `close`}>{section.content}</p>
                    </li>   
                )
            })
        )
    }
}

export default Accordion