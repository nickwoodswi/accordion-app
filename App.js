import React from 'react'
import './sections.js'
import Accordion from './Accordion'

function App() {
    return(
        <ul>
            <Accordion />
        </ul>
    )
}

export default App