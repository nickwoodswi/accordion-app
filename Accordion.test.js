// make React available
import React from 'react';
import renderer from 'react-test-renderer';
import Accordion from './Accordion'

// make the ReactDOM available, necessary for rendering the component
import ReactDOM from 'react-dom';

// make the App component available
import App from './App';

// this is the test case
it('renders without crashing', () => {
  // first create a DOM element to render the component into
  const div = document.createElement('div');

  // render the component, this is the actual test, if something is wrong it will fail here
  ReactDOM.render(<App />, div);

  // clean up code
  ReactDOM.unmountComponentAtNode(div);
});

it('renders Accordion without crashing', () => {
    // first create a DOM element to render the component into
    const div = document.createElement('div');
  
    // render the component, this is the actual test, if something is wrong it will fail here
    ReactDOM.render(<Accordion />, div);
  
    // clean up code
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the UI as expected', () => {
    const tree = renderer
      .create(<Accordion />)
      .toJSON();
    expect(tree).toMatchSnapshot();  
    });