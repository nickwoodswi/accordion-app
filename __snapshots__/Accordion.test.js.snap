// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
Array [
  <li>
    <button
      onClick={[Function]}
    >
      Section 1
    </button>
    <p
      className="close"
    >
      Lorem ipsum dolor sit amet consectetur adipisicing elit.
    </p>
  </li>,
  <li>
    <button
      onClick={[Function]}
    >
      Section 2
    </button>
    <p
      className="close"
    >
      Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!
    </p>
  </li>,
  <li>
    <button
      onClick={[Function]}
    >
      Section 3
    </button>
    <p
      className="close"
    >
      Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?
    </p>
  </li>,
]
`;
